### PR TITLE
subsys/shell/mqtt: Handle retval of `zsock_poll`

### PR DESCRIPTION
**Pull request by ycsin**
1. Some MISRA-related fixes
2. Handle retval of `zsock_poll` properly

### CODE REVIEW

**Review by jukkar — DISMISSED** [review]

File: subsys/shell/backends/shell_mqtt.c [hunk 1]
```diff
@@ -124,9 +124,12 @@ static int wait(int timeout)
 static int get_mqtt_broker_addrinfo(void)
 {
 	int rc;
-	struct zsock_addrinfo hints = { .ai_family = AF_INET,
-					.ai_socktype = SOCK_STREAM,
-					.ai_protocol = 0 };
+	struct zsock_addrinfo hints;
+
+	(void)memset(&hints, 0, sizeof(hints));
```
Comment (rlubos) [inline subsys/shell/backends/shell_mqtt.c:136]: I might be missing something, but is this change actually fixing anything? As `hints` structure has an initializer, remainig, uninitialized fields should be zeroed anyway.

**Review by rlubos — COMMENTED** [review]
Looks mostly good, but please drop the `subsys` part from the commit titles, it's disallowed (not sure why wasn't it caught by the CI).

File: subsys/shell/backends/shell_mqtt.c (hunk 1 above)
Comment (ycsin) [reply subsys/shell/backends/shell_mqtt.c:136]: The variable is on stack, so if not `memset` explicitly I guess there might be some random values for the remaining fields?

Comment (rlubos) [reply subsys/shell/backends/shell_mqtt.c:136]: That is true only if there were no initialization at all, then a stack variable would contain rubbish. But if there is an initializer (may even be empty), all of the remaining fields are zeroed, it's guaranteed by the standard.

Comment (ycsin) [reply subsys/shell/backends/shell_mqtt.c:136]: Interesting, that's something new for me, I will change them

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**